src/bridge/cockpitfsdir.c

```diff
@@ -121,6 +127,8 @@ on_files_listed (GObject *source_object,
       json_object_set_string_member (msg, "event", "present");
       json_object_set_string_member
         (msg, "path", g_file_info_get_attribute_byte_string (info, G_FILE_ATTRIBUTE_STANDARD_NAME));
+      json_object_set_int_member
+        (msg, "type", g_file_info_get_file_type (info));
```

Comment:
Yes, this is necessary so that the protocol has the possibility of being used by another Unix besides Linux.